rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java

```diff
@@ -97,16 +104,12 @@ private void registerCoordinatorService(final OperationContext context, final Mo
         final String socketBindingName = model.get(Attribute.SOCKET_BINDING.getLocalName()).asString();
         final String serverName = model.get(Attribute.SERVER.getLocalName()).asString();
         final String hostName = model.get(Attribute.HOST.getLocalName()).asString();
-        final CoordinatorService coordinatorService = new CoordinatorService();
-        final ServiceBuilder<CoordinatorService> coordinatorServiceBuilder = context
-                .getServiceTarget()
-                .addService(RTSSubsystemExtension.COORDINATOR, coordinatorService)
-                .addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBindingName), SocketBinding.class,
-                        coordinatorService.getInjectedSocketBinding())
-                .addDependency(UndertowService.virtualHostName(serverName, hostName), Host.class,
-                        coordinatorService.getInjectedHost());
-
-        coordinatorServiceBuilder
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(RTS_COORDINATOR_CAPABILITY);
+        Supplier<Host> hostSupplier = builder.requires(UndertowService.virtualHostName(serverName, hostName));
```

Comment:
This should also use a capability reference, but we can address separately.  Can you file a jira?

Reply:
Created https://issues.redhat.com/browse/WFLY-17882